### PR TITLE
Complex_Calculator 258092

### PR DESCRIPTION
**Pull request by 28-shravya**
The functions are added and modified in src, inc folders.

### CODE REVIEW

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c [hunk 1]
```diff
@@ -25,6 +27,18 @@ void test_zero(void)
     TEST_ASSERT_EQUAL(SUCCESS, complex_sum(&c1, &c2, &result));
     TEST_ASSERT_EQUAL(0, result.real);
     TEST_ASSERT_EQUAL(0, result.imaginary);
+
```
Comment (Bharathgopal) [inline Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:30]: Write different test cases for each function. It is more readable and also easy to find the bugs.

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c [hunk 2]
```diff
@@ -46,11 +72,23 @@ void test_negative(void)
     
     c2.real = 75;
     c2.imaginary = 95;
-    
+
     TEST_ASSERT_EQUAL(SUCCESS, complex_sum(&c1, &c2, &result));
     TEST_ASSERT_EQUAL(50, result.real);
     TEST_ASSERT_EQUAL(80, result.imaginary);
 
+    TEST_ASSERT_EQUAL(SUCCESS, complex_diff(&c1, &c2, &result));
```
Comment (Bharathgopal) [inline Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:80]: Write different test cases for each function written. It is more readable and also easy to find the bugs.

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c [hunk 3]
```diff
@@ -61,6 +99,18 @@ void test_negative(void)
...
 
+    TEST_ASSERT_EQUAL(SUCCESS, complex_diff(&c1, &c2, &result));
+    TEST_ASSERT_EQUAL(50, result.real);
+    TEST_ASSERT_EQUAL(80, result.imaginary);
+    
+    TEST_ASSERT_EQUAL(SUCCESS, complex_mul(&c1, &c2, &result));
+    TEST_ASSERT_EQUAL(450, result.real);
+    TEST_ASSERT_EQUAL(3500, result.imaginary);
+    
+    TEST_ASSERT_EQUAL(SUCCESS, complex_div(&c1, &c2, &result));
```
Comment (Bharathgopal) [inline Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:110]: Consider testing with more values. Ex: all values are 0, or only imaginary part is 0. All corner cases.

**Comment by Bharathgopal** [comment]
Good implementation with few changes required. Do the suggested changes and add only the link to your repo under the **README.md** file and Create a PR.

Thank you
Bharath

**Review by Bharathgopal — CHANGES_REQUESTED** [review]

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c (hunk 1 above)
Comment (28-shravya) [reply Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:30]: Sure sir I will do it.

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c (hunk 2 above)
Comment (28-shravya) [reply Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:80]: Sure sir. I will do it.

File: Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c (hunk 3 above)
Comment (28-shravya) [reply Example_Programs/programming_concpets/calculator_complex/test/test_complex_calculator.c:110]: okay sir. I will write more test cases for each operations.

**Comment by 28-shravya** [comment]
> Good implementation with few changes required. Do the suggested changes and add only the link to your repo under the **README.md** file and Create a PR.
> 
> Thank you
> Bharath

**Comment by Bharathgopal** [comment]
Closing this PR as I have added your solution to README file.
